src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -19,7 +19,7 @@ internal sealed class SymbolFactory : SymbolFactory<PEModuleSymbol, TypeSymbol>
                 return elementType;
             }
 
-            return ArrayTypeSymbol.CreateMDArray(moduleSymbol.ContainingAssembly, CreateType(elementType, customModifiers), rank, sizes, lowerBounds);
+            return ArrayTypeSymbol.CreateMDArray(moduleSymbol.ContainingAssembly, CreateType(moduleSymbol, elementType, customModifiers), rank, sizes, lowerBounds);
```

Comment:
I assume whatever NonNullTypes value we pass here will not be observable.
If so, maybe we should pass in a constant.

Reply:
Then I wonder if we should be using the NonNullTypes context from the module, or rather a more specific context.

Reply:
This still doesn't feel right. This method is used for decoding types from metadata. For instance from a signature of a method.
Then I would expect the `NonNullTypes` context from the method to apply.
That will be done by running `ApplyNullableTransforms` on the decoded types. So the `NonNullTypes` used by the metadata decoder should not matter.
Generally, I don't think we should use the `NonNullTypes` from the module directly. We should always be asking from a more specific symbol, which will look up the chain and ultimately fall back to the module.